x-pack/plugins/infra/server/lib/metrics/lib/calculate_bucket_size/unit_to_seconds.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { sortBy, isNumber } from 'lodash';
+import { INTERVAL_STRING_RE } from './interval_regex';
+
+export const ASCENDING_UNIT_ORDER = ['ms', 's', 'm', 'h', 'd', 'w', 'M', 'y'];
+
+const units: Record<string, number> = {
+  ms: 0.001,
+  s: 1,
+  m: 60,
+  h: 3600,
+  d: 86400,
+  w: 86400 * 7, // Hum... might be wrong
+  M: 86400 * 7 * 4, // this too... 29,30,31?
```

Comment:
Moment.js uses 30 days for a month and it looks like `elastic-datemath` uses that too